packages/stateManager/index.ts

```diff
@@ -615,10 +628,47 @@ export class StateManager {
       this.keycards,
     ];
     const stream = this.client.createEventStream();
+    const shopClient = createPublicClient({
+      chain: process.env.DEV ? hardhat : mainnet,
+      transport: http(),
+    });
+    //When we inititally create a shop, we are saving the relay tokenId => shopId.
+    //Here, we are retrieving all the relay addresses associated with the shopId and saving them to keycards store.
+    //Since some shopEvents are signed by a relay, we need to include these addresses when verifying the event signer.
+    const count = (await shopClient.readContract({
+      address: abi.addresses.ShopReg as `0x${string}`,
+      abi: abi.ShopReg,
+      functionName: "getRelayCount",
+      args: [this.shopId],
+    })) as number;
+
+    if (count > 0) {
+      const tokenIds = (await shopClient.readContract({
+        address: abi.addresses.ShopReg as `0x${string}`,
+        abi: abi.ShopReg,
+        functionName: "getAllRelays",
+        args: [this.shopId],
+      })) as `0x${string}`[];
+      for await (const tokenId of tokenIds) {
+        const ownerAdd = (await shopClient!.readContract({
+          address: abi.addresses.RelayReg as `0x${string}`,
+          abi: abi.RelayReg,
+          functionName: "ownerOf",
+          args: [tokenId],
+        })) as `0x${string}`;
+        await this.keycards.addRelayAddress(ownerAdd);
+      }
+    }
+
     //Each event will go through all the storeObjects and update the relevant stores.
     for await (const event of stream) {
-      for (const storeObject of storeObjects) {
-        await storeObject._processEvent(event.event);
+      try {
+        const wa = await this.keycards.get(event.signer.toLowerCase());
+        for (const storeObject of storeObjects) {
+          await storeObject._processEvent(event.event);
+        }
+      } catch (error) {
+        console.log(`Event not verified: ${(event.signer, event.event)}`);
```

Comment:
hmmm.. these would vanish silently for most users but I'm not sure what exactly we _do_ want to happen here. cc @wanderer

Reply:
Maybe this should boil up to the UI? Which mean we would not catch it here.

Reply:
remove the try catch and write test that halts the operation